### ISSUE TITLE
ArrayIndexOutOfBoundsException on input with multiple bytes for a character

### ISSUE THREAD

**Issue by ktulinger** [f4b6a3/ulid-creator#32]
When using 

```
Ulid.from(String)
```

it fails with 

```
java.lang.ArrayIndexOutOfBoundsException: Index 3617 out of bounds for length 256
	at com.github.f4b6a3.ulid.Ulid.isValidCharArray(Ulid.java:801)
	at com.github.f4b6a3.ulid.Ulid.toCharArray(Ulid.java:776)
	at com.github.f4b6a3.ulid.Ulid.from(Ulid.java:325)
```

when the input string contains characters represented by more than 1 byte - in this example 3617. I had a string of length 26 java `char` however some of them are 2 bytes, rather than 1 (java `char` type is 2B). I'd expect the library to handle the input validity correctly and consistently throw `IllegalArgumentException` on invalid input.

**Comment by fabiolimace** [comment]
The validator did not expect multibyte characters.

I changed the validation method to catch `ArrayIndexOutOfBoundsException` and return `false` (invalid) so that `IllegalArgumentException` is thrown instead.

See this commit https://github.com/f4b6a3/ulid-creator/commit/733242f9800eebe84436deaee4eaa415bafda8f6.

**Comment by fabiolimace** [comment]
Released v5.2.3! 🎉

**Comment by ktulinger** [comment]
Thank you for the quick turnaround. I believe checking the array boundaries instead of try-catch is preferred approach overall. But thanks anyway.
